src/runtime/CMakeLists.txt

```diff
@@ -168,20 +169,52 @@ foreach (i IN LISTS RUNTIME_CPP)
     foreach (j IN ITEMS 32 64)
         # -fpic needs special treatment; see below on windows 64bits
         set(fpic -fpic)
-        if (j EQUAL 32)
-            if (i MATCHES "windows_.*")
-                # win32 uses the stdcall calling convention, which is x86-specific
-                set(TARGET "i386-unknown-unknown-unknown")
-            else ()
+        # for the generic windows 64-bit target, we need -fshort-wchar
+        set(fshort-wchar "")
+        # Windows
+        if (i MATCHES "windows_.*")
+            # Windows x86/x64
+            if (i MATCHES "windows_.*_x86$")
+                if (j EQUAL 32)
+                    # win32 uses the stdcall calling convention, which is x86-specific
+                    set(fpic "")
+                    set(TARGET "i386-unknown-windows-unknown")
+                else ()
+                    # must omit -fpic, otherwise clang will complain with the following:
+                    # clang : error : unsupported option '-fpic' for target 'x86_64-pc-windows-msvc'
+                    set(fpic "")
+                    set(TARGET "x86_64-unknown-windows-unknown")
+                endif ()
+            # Windows on ARM
+            elseif (i MATCHES "windows_.*_arm$")
+                if (j EQUAL 32)
+                    set(fpic "")
+                    set(TARGET "arm-unknown-windows-unknown")
+                else ()
+                    set(fpic "")
+                    set(TARGET "aarch64-unknown-windows-unknown")
+                endif ()
+            # Windows Generic
+            else()
+                if (j EQUAL 32)
+                    # TODO(marcos): generic code won't hold for ARM32... If ARM32 support becomes necessary,
```

Comment:
Do we need to support Windows-for-ARM32 (now or ever)?

Reply:
I don't think so. We can worry about it when (and if) it happens. ARM64 is way more relevant here.
(And I don't think there is any Windows on ARM 32 device out there that supports D3D12)